tests/syntax/test_equals.py

```diff
@@ -0,0 +1,228 @@
+from __future__ import unicode_literals
+import unittest
+import sys
+
+sys.path.append('.')
+
+from tests.syntax import dedent_ftl
+from fluent.syntax.parser import FluentParser
+
+
+class TestEntryEqualToSelf(unittest.TestCase):
+    def setUp(self):
+        self.parser = FluentParser()
+
+    def parse_ftl_entry(self, string):
+        return self.parser.parse_entry(dedent_ftl(string))
+
+    def test_same_simple_message(self):
+        message1 = self.parse_ftl_entry("""\
+            foo = Foo
+        """)
+
+        self.assertTrue(message1.equals(message1))
+
+    def test_same_selector_message(self):
+        message1 = self.parse_ftl_entry("""\
+            foo =
+                { $num ->
+                    [one] One
+                    [two] Two
+                    [few] Few
+                    [many] Many
+                   *[other] Other
+                }
+        """)
+
+        self.assertTrue(message1.equals(message1))
```

Comment:
Does it make sense to test for equality against the identity traversal, too?

If so, add that to the tests in this class?